### PR TITLE
add Table to Bookings

### PR DESCRIPTION
**Pull request by rebecalsantos**
-Create new table that includes desk tables. Populate table to 6.
- update POST request to accept an optional desk_id.
- add desks to database/backend GET request

### CODE REVIEW

File: server/database_operations.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("pg");
+const config = require("../config");
```
Comment (gregdyke) [inline server/database_operations.js]: I don't think we have a config

File: server/database_operations.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("pg");
```
Comment (gregdyke) [inline server/database_operations.js:1]: should probably import db, as that contains a connexion string

File: server/database_operations.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const pool = new Pool(config);
+
+queryDb(createSchema)
+  .then(() => console.log("schema created successfully"))
+  .then(() => queryDb(populateDb))
+  .then(() => console.log("data populated successfully"))
+  .then(() => process.exit(0));
+
+function queryDb(query) {
+  return pool.query(query).catch((error) => {
```
Comment (gregdyke) [inline server/database_operations.js:20]: then here can be db.query

File: server/api.js [hunk 4]
```diff
@@ -24,20 +24,26 @@ router.get("/bookings", async (req, res) => {
 
 router.post("/bookings", async function (req, res) {
 	const userName = req.body.name;
+	const tableName = req.body.table;
```
Comment (gregdyke) [inline server/api.js]: desk (table would be confusing :D)
and I think it can pass in the desk_id

File: server/api.js [hunk 5]
```diff
@@ -24,20 +24,26 @@ router.get("/bookings", async (req, res) => {
...
 
 	const insertQuery =
-		"INSERT INTO booking (desk_user_id, booking_date) VALUES ($1, $2)";
+		"INSERT INTO booking (desk_user_id, desk_table, booking_date) VALUES ($1, $2, $3)";
 	try {
 		const userResult = await db.query(
 			"SELECT id from desk_user WHERE username=$1",
 			[userName]
 		);
+		const tableResult = await db.query(
```
Comment (gregdyke) [inline server/api.js]: if we pass in the desk_id, we won'd need this query

File: server/api.js [hunk 6]
```diff
@@ -24,20 +24,26 @@ router.get("/bookings", async (req, res) => {
...
+			"SELECT id from desk_table WHERE desk_name=$2",
+			[tableName]
+		);
 		if (userResult.rows.length === 0) {
 			return res.status(400).send(`User with name ${userName} does not exist`);
 		}
 		const userId = userResult.rows[0].id;
-		const bookingResult = await db.query(insertQuery, [userId, bookingDate]);
+		const tableId = tableResult.rows[0].id;
```
Comment (gregdyke) [inline server/api.js]: (you didn't check your tableResult - but no matter - don't need it)

File: server/recreate_squema.sql [hunk 7]
```diff
@@ -0,0 +1,15 @@
+drop table if exists booking;
+drop table if exists desk_table;
+drop table if exists desk_user;
+create table desk_user (id SERIAL PRIMARY KEY, username VARCHAR(200) not null);
+create table desk_table (
```
Comment (gregdyke) [inline server/recreate_squema.sql]: should be called desk, I think

File: server/api.js [hunk 8]
```diff
@@ -13,7 +13,7 @@ router.get("/bookings", async (req, res) => {
 		);
 		// convert array with objects having {username, booking_date} into {name,date}
 		const bookings = result.rows.map((row) => {
-			return { id: row.id, name: row.username, date: row.booking_date };
+			return { id: row.id, name: row.username, table: row.desk_table, date: row.booking_date };
```
Comment (gregdyke) [inline server/api.js]: the query should join against desk, and we should use both the desk_id and the deskname (as per the trello ticket)

**Review by gregdyke — CHANGES_REQUESTED** [review]
Very nice start! I like that you changed both POST and GET, though I'm not sure you made them quite right :)

**Comment by gregdyke** [comment]
(also, needs the change to package.json to add the script, but that could be done in a followup

File: server/database_operations.js (hunk 3 above)
Comment (rebecalsantos) [reply server/database_operations.js:20]: I'm not too sure if I understand this

File: server/database_operations.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("db");
+const config = require("./config");
```
Comment (teomrd) [inline server/database_operations.js]: I think in here you need

```js
const { config } = require("./config");
```

as you are exported as a named export (_not default export_)

File: server/database_operations.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("db");
```
Comment (teomrd) [inline server/database_operations.js]: And that should be

```js
import { Pool } from "pg";
```

As Pool is a Postgres module

File: server/recreate_schema.sql [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+drop table if exists desk;
+drop table if exists desk_user;
+create table desk_user (id SERIAL PRIMARY KEY, username VARCHAR(200) not null);
+create table desk (
+    id SERIAL PRIMARY KEY,
+    desk_name VARCHAR (100) not null);
+create table booking (
+    id SERIAL PRIMARY KEY,
+    desk_user_id INT REFERENCES desk_user(id),
+    desk_id INT REFERENCES desk(id)
```
Comment (teomrd) [inline server/recreate_schema.sql]: Seems that you are missing a comma "," in here

**Review by rebecalsantos — COMMENTED** [review]
made changes. deskId doesn't save though

File: server/database_operations.js (hunk 10 above)
Comment (gregdyke) [reply server/database_operations.js]: ah, that works too. I thought we would just import db from "./db"

File: server/database_operations.js (hunk 3 above)
Comment (gregdyke) [reply server/database_operations.js:20]: that's fine. With the approach with config.js, you can ignore my comments on this file

File: server/api.js [hunk 12]
```diff
@@ -11,9 +11,9 @@ router.get("/bookings", async (req, res) => {
 	try {
 		const result = await db.query("SELECT booking.*,desk_user.username from booking INNER JOIN desk_user ON booking.desk_user_id=desk_user.id;"
```
Comment (gregdyke) [inline server/api.js]: I think this query needs editing to also  inner join against desk, in order to get the desk_name

File: server/populate_db.sql [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+INSERT INTO desk (desk_name) VALUES ('desk 4');
+INSERT INTO desk (desk_name) VALUES ('desk 5');
+INSERT INTO desk (desk_name) VALUES ('desk 6');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (1,'2022-01-17'); 
+INSERT INTO booking (desk_user_id, booking_date) VALUES (2,'2022-01-19');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (1,'2022-01-19');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (3,'2022-01-19');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (3,'2022-01-20');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (1,'2022-01-20');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (1,'2022-01-21');
```
Comment (gregdyke) [inline server/populate_db.sql]: This file is missing the new, desk based bookings

File: server/api.js [hunk 14]
```diff
@@ -24,20 +24,27 @@ router.get("/bookings", async (req, res) => {
 
 router.post("/bookings", async function (req, res) {
 	const userName = req.body.name;
+	const deskId = req.body.deskId;
```
Comment (gregdyke) [inline server/api.js]: desk_id, probably for consistency with the other filed names, and with the objects returned by the GET request (sorry, naming things well is hard, but worth doing :) )

**Review by gregdyke — CHANGES_REQUESTED** [review]

**Review by gregdyke — APPROVED** [review]

**Pull request merged**